src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageSearch/Table.cs

```diff
@@ -4,119 +4,232 @@
+    }
+
     internal class Table
     {
+        // This is the default window width if we cannot get the actual window width
+        const int DefaultWindowWidth = 115;
+        // This is the minimum number of characters in a column which includes "| c |" where c is a character
+        const int MinimumCharactersInAColumn = 4;
+        // This is the list of columns in the table
+        internal readonly List<Column> _columns = new List<Column>();
+        // This is the list of rows in the table
         internal List<string[]> _rows = new List<string[]>();
-        private int[] _columnWidths;
+        // This is the list of columns to highlight
         private int[] _columnsToHighlight;
+        // This is the highlighter color
+        private ConsoleColor _highlighter = ConsoleColor.Red;
+        // This is the maximum column width: The maximum number of characters in a column based on the window width
+        private readonly int _maxColumnWidth;
+        // This is the default console color
+        private readonly ConsoleColor _consoleColor = Console.ForegroundColor;
 
         public Table(int[] columnsToHighlight, params string[] headers)
         {
             _columnsToHighlight = columnsToHighlight;
-            _columnWidths = new int[headers.Length];
+            int windowWidth = -1;
 
-            for (int i = 0; i < headers.Length; i++)
+            // Get the window width if possible 
+            try
             {
-                _columnWidths[i] = headers[i].Length;
+                windowWidth = Console.WindowWidth;
+            }
+            catch (Exception)
+            {
+                // Ignore any exception
             }
 
-            _rows.Add(headers);
+            // If the window width is not available, use the default window width
+            if (windowWidth <= 0)
+            {
+                _maxColumnWidth = DefaultWindowWidth;
+            }
+            else
+            {
+                _maxColumnWidth = Math.Max(MinimumCharactersInAColumn, (windowWidth - MinimumCharactersInAColumn * headers.Length) / headers.Length);
+            }
+
+            // Add the headers
+            foreach (var header in headers)
+            {
+                _columns.Add(new Column { Header = header, Width = header.Length });
+            }
         }
 
+        /* Add a row to the table
+         * row: The list of values in the row
+         */
         public void AddRow(params string[] row)
         {
-            if (row.Length != _columnWidths.Length)
+            if (row.Length != _columns.Count)
             {
                 throw new InvalidOperationException("Row column count does not match header column count.");
             }
 
             for (int i = 0; i < row.Length; i++)
             {
-                _columnWidths[i] = Math.Max(_columnWidths[i], row[i]?.Length ?? 0);
+                _columns[i].Width = Math.Min(_maxColumnWidth, Math.Max(_columns[i].Width, row[i]?.Length ?? 0));
```

Comment:
Try to understand the algorithm here, the order of columns would guide how much space is taken correct?

Reply:
just updated the description to contain the algorithm. Please let me know if you have any questions